server/routes/auth.js

```diff
@@ -60,7 +63,8 @@ module.exports = function(app) {
 		}
 	});
 
-	app.post('/api/register', async (req, res) => {
+	//second parameter is uploading a single pic with multer (the profile pic):
+	app.post('/api/register', upload.single('profilePic'), async (req, res) => {
```

Comment:
This shouldn't be part of the register route.
You need to create 3 separate routes for the profile pic

GET /api/me/avatar
POST /api/me/avatar
DELETE /api/me/avatar

It's a completely separate call

Reply:
if we store the binary data on the user an example for a GET call would look something like this: 

app.get('/api/me/avatar', async (req, res) => {
		try {
			const user = await getUser(req.user.email);
			if (!user || !user.avatar) {
				throw new Error();
			}
			res.set('Content-Type', 'image/jpeg'); // <----- this is possible because of [sharp](https://www.npmjs.com/package/sharp), we can set a fixed format for images
			res.send(user.avatar);
		} catch (e) {
			res
				.status(404)
				.json({ message: 'Either the user or the avatar do not exist' })
				.end();
		}
	});